server/base/src/main/java/org/apache/accumulo/server/fs/PreferredVolumeChooser.java

```
@@ -21,84 +21,90 @@
+    }
+    ServerConfigurationFactory localConf = loadConf();
 
-    // Get the current table's properties, and find the preferred volumes property
-    // This local variable is an intentional component of the single-check idiom.
-    ServerConfigurationFactory localConf = serverConfs;
-    if (localConf == null) {
-      // If we're under contention when first getting here we'll throw away some initializations.
-      localConf = new ServerConfigurationFactory(HdfsZooInstance.getInstance());
-      serverConfs = localConf;
+    String systemPreferredVolumes = localConf.getConfiguration().get(PREFERRED_VOLUMES_CUSTOM_KEY);
+    if (null == systemPreferredVolumes || systemPreferredVolumes.isEmpty()) {
+      String logMessage = "Default preferred volumes are missing.";
+      log.debug(logMessage);
+      throw new AccumuloException(logMessage);
     }
-    TableConfiguration tableConf = localConf.getTableConfiguration(env.getTableId());
-    final Map<String,String> props = new HashMap<>();
-    tableConf.getProperties(props, PREFERRED_VOLUMES_FILTER);
-    if (props.isEmpty()) {
-      log.warn("No preferred volumes specified. Defaulting to randomly choosing from instance volumes");
-      return super.choose(env, options);
+
+    String volumes = null;
+    if (env.hasTableId()) {
+      volumes = localConf.getTableConfiguration(env.getTableId()).get(PREFERRED_VOLUMES_CUSTOM_KEY);
+    } else if (env.hasScope()) {
+      volumes = localConf.getConfiguration().get(Property.GENERAL_ARBITRARY_PROP_PREFIX.getKey() + env.getScope() + ".preferredVolumes");
+    }
+
+    // if there was an empty or missing property, use the system-wide volumes
+    if (null == volumes || volumes.isEmpty()) {
+      if (env.hasTableId()) {
+        log.warn("Missing property for TableID " + env.getTableId() + " but it should have picked up default volumes.");
+      } else {
+        log.debug("Missing preferred volumes for scope " + env.getScope() + ". Using default volumes.");
```

Why is this debug?  Should the expected property name be in the message?  Should this fail instead of log?